quarkchain/cluster/tests/test_jsonrpc.py

```diff
@@ -57,6 +65,57 @@ def send_request(*args):
     return call_async(__send_request(*args))
 
 
+@contextmanager
+def jrpc_websocket_server_context(slave_server):
+    env = DEFAULT_ENV.copy()
+    env.cluster_config = ClusterConfig()
+    env.cluster_config.JSON_RPC_PORT = 38391
+    # to pass the circleCi
+    env.cluster_config.JSON_RPC_HOST = "127.0.0.1"
+
+    slave_config = SlaveConfig()
+    slave_config.HOST = "0.0.0.0"
+    slave_config.PORT = 38001
+    slave_config.WEBSOCKET_JSON_RPC_PORT = 38591
+    slave_config.ID = "S1"
+    slave_config.CHAIN_MASK_LIST = [5]
+
+    env.cluster_config.SLAVE_LIST.append(slave_config)
+    env.slave_config = env.cluster_config.get_slave_config("S1")
+    server = JSONRPCWebsocketServer.start_websocket_server(env, slave_server)
+    try:
+        yield server
+    finally:
+        server.shutdown()
+
+
+def send_websocket_request(request):
```

Comment:
maybe this?
https://jsonrpcclient.readthedocs.io/en/latest/examples.html#websockets

![image](https://user-images.githubusercontent.com/38544039/63401238-e261f900-c3d6-11e9-9eee-b9d87cf101e1.png)